### PR TITLE
MAGECLOUD-2681 Explain Docker

### PR DESCRIPTION
**Pull request by hguthrie**
- Update cloud navigation to highlight Docker environment
- Update Local setup to be Local Development, added two paths: Docker setup and Manual setup
- Split Docker Launch from the Docker environment architecture
- Added Common Docker command list, including how to invoke ece-tools and magento-cloud

Do not MERGE until the following is complete:
- [x] Technical assist and review (Some info is missing and requires additional help)
- [ ] Link check
- [x] End of Sprint approval
- [x] Peer edit

whatsnew
Renamed Local environment setup to [Local development](). Added new [Docker development]() architecture.

### CODE REVIEW

File: guides/v2.1/cloud/docker/docker-development.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+-  **FPM container**—[magento/magento-cloud-docker-php](https://hub.docker.com/r/magento/magento-cloud-docker-php/)  
+    PHP-CLI: version 7 and later  
+    PHP-FPM: version 7 and later  
+-  **DB container**  
+    {{site.data.var.ece}} uses MariaDB for the local database. The DB container, based on MariaDB v10, 
+-  **Redis container**—[magento/magento-cloud-docker-redis](https://hub.docker.com/r/magento/magento-cloud-docker-redis/)  
+    Based on the latest Redis version,
+-  **Varnish container**—[magento/magento-cloud-docker-varnish](https://hub.docker.com/r/magento/magento-cloud-docker-varnish/)  
+    Based on the lates Varnish version, it is used for front cache. It sends requests to web container and caches result.
```
Comment (shiftedreality) [inline guides/v2.1/cloud/docker/docker-development.md]: lates => latest

File: guides/v2.1/cloud/docker/docker-development.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+-  **Web container**—[magento/magento-cloud-docker-nginx](https://hub.docker.com/r/magento/magento-cloud-docker-nginx/)  
+    {{site.data.var.ece}} uses NGINX for the local web server. The web container, based on NGINX v1.9, mainly works with the [PHP-FPM](https://php-fpm.org) to serve PHP code and the DB container.
```
Comment (shiftedreality) [inline guides/v2.1/cloud/docker/docker-development.md]: remove `mainly`, we don't have alternative options

File: guides/v2.1/cloud/docker/docker-development.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+### CLI container
+
+The Docker container architecture for the {{site.data.var.ece}} platform includes a **CLI** container, which is based on a 7.0-cli or 7.1-cli image, provides {{site.data.var.ece}} and `{{site.data.var.ct}}` commands and performs file system operations. The CLI container depends on the DB container and the Redis container.
+
+-  `build`—extends the CLI container to perform ??
```
Comment (shiftedreality) [inline guides/v2.1/cloud/docker/docker-development.md]: > to perform operations with writable filesystem similar to `build` phase of Magento Cloud

File: _data/toc/cloud-guide.yml [hunk 4]
```diff
@@ -74,47 +71,50 @@ pages:
...
 
-    - label: Local environment setup
+    - label: Local development
       url: /cloud/access-acct/first-time-setup.html
       children:
-        - label: Prepare for local environment setup
-          url: /cloud/before/before-workspace.html
+        - label: Docker development
+          url: /cloud/docker/docker-development.html
+          include_versions: ["2.1","2.2","2.3"]
```
Comment (belbiy) [inline _data/toc/cloud-guide.yml]: If the menu item is supposed to be for every version instead of 2.0, i suggest using `exclude_versions: "2.0"`.

Comment (hguthrie) [reply _data/toc/cloud-guide.yml]: Yes! I need to update with new TOC strategy! Thanks.

**Comment by mveeramneni** [comment]
Are these instructions specific to MAC or do they work same for Windows machine?

**Comment by hguthrie** [comment]
I don't know that DevDocs posts Cloud instructions for Windows. I suppose we could. I'll ask about it.

**Comment by hguthrie** [comment]
@mveeramneni  MAGECLOUD-2728 created to document Docker environment for Windows users.

**Comment by mveeramneni** [comment]
Thank you @hguthrie, I think this will help windows users

**Pull request merged**